linux-installer/storage/config_helper.go

```diff
@@ -0,0 +1,402 @@
+        os.O_CREATE | os.O_RDWR | os.O_TRUNC,
+        0644)
+    if err != nil {
+        panic("Error opening ./config.json, please check filesystem permissions and disk space availability")
+    }
+    defer cfile.Close()
+    b, _ := json.MarshalIndent(configObj,"","    ")
+    cfile.Write(b)
+    cfile.Write([]byte("\n"))
+}
+
+/*
+* Create a valid config file and even set up a local cassandra if necessary
+*/
+func main() {
+    var (
+        configType      string
+        configObj       config
+    )
+    // read command line arguments
+    flag.StringVar(&configType, "config", "cluster", "The type of the config to create.")
+    flag.Parse()
+    
+    // Gather database/storage settings
+    switch configType {
+        case "local":
+            configObj.Storage = "cassandra"
+            configObj.ObjStorage = "local-fs"
+            configObj.Database = get_cassandra_boxes(1)
+            err := setup_cassandra_keyspace(configObj.Database)
+            if err != nil {
+                panic(err)
+            }
+            
+        case "local-objstorage":
+            configObj.Storage = "cassandra"
+            configObj.ObjStorage = "local-fs"
+            configObj.Database = get_cassandra_boxes(-1)  // unlimited boxes
+            err := setup_cassandra_keyspace(configObj.Database)
+            if err != nil {
+                panic(err)
+            }
+            
+        case "local-cassandra":
+            configObj.Storage = "cassandra"
+            configObj.ObjStorage = "S3"
+            configObj.Database = get_cassandra_boxes(1)
+            configObj.ObjDatabase = get_s3_boxes(1)  // current limit seems to be 1 (see objStorerS3.go)
+            err := setup_cassandra_keyspace(configObj.Database)
+            if err != nil {
+                panic(err)
+            }
+            
+        case "cluster":
+            configObj.Storage = "cassandra" 
+            configObj.ObjStorage = "S3"
+            configObj.Database = get_cassandra_boxes(-1)  // unlimited boxes
+            configObj.ObjDatabase = get_s3_boxes(1)  // current limit seems to be 1 (see objStorerS3.go)
+            err := setup_cassandra_keyspace(configObj.Database)
+            if err != nil {
+                panic(err)
+            }
+            
+        case "local-mongodb":
+            configObj.Storage = "mongodb"
+            configObj.Database = get_mongodb_boxes(1)
+            
+        case "cluster-mongodb":
+            configObj.Storage = "mongodb"
+            configObj.Database = get_mongodb_boxes(-1)  // unlimited boxes
+            
+        default:
+            fmt.Println("Invalid option for --config= supplied.")
+            panic("Invalid option for --config= supplied.")
+    }
+    
+    // gather various other information
+    fmt.Println("> Other settings:")
+    configObj.LogFile       = get_text(msg_logfile, "")
+    configObj.LogLevel      = get_text_generic(msg_loglevel, "debug", "error/warn/info/debug")
```

Comment:
only supported log levels are "warning/info/debug", anything else will default to "debug"